src/bentoml/_internal/io_descriptors/base.py

```diff
@@ -64,27 +66,51 @@ def openapi_responses(self) -> dict[str, t.Any]:
         raise NotImplementedError
 
 
+DEFAULT_FROM_SAMPLE_DOCSTRING = """\
+Creates an instance of the :class:`{name}` from given sample. Note that ``from_sample`` does not
+take positional arguments.
+
+Args:
+    sample: The sample to create the instance from.
+    kwargs: Additional keyword arguments to pass to the constructor.
+
+Returns:
+    :class:`{name}`: The instanc
+"""
+
+
 class IODescriptor(ABC, _OpenAPIMeta, t.Generic[IOType]):
     """
     IODescriptor describes the input/output data format of an InferenceAPI defined
     in a :code:`bentoml.Service`. This is an abstract base class for extending new HTTP
     endpoint IO descriptor types in BentoServer.
     """
 
-    __slots__ = ("_args", "_kwargs", "_proto_fields", "_mime_type", "descriptor_id")
+    if TYPE_CHECKING:
+        # Populated by subclasses. Makes IDE happy.
```

Comment:
What IDE is the problem, and what's the error?

Reply:
pyright.